app/Livewire/Project/Service/TemplateUpdate.php

```
@@ -0,0 +1,192 @@
+
+        $accepted = array_map('intval', array_keys(array_filter($this->acceptedHunks)));
+
+        return ComposeDiff::apply((string) $this->service->docker_compose_raw, $this->latestCompose, $accepted);
+    }
+
+    private function persistCompose(string $compose): void
+    {
+        try {
+            $this->authorize('update', $this->service);
+
+            if (trim($compose) === '' || ! ComposeDiff::isValidYaml($compose)) {
```

PHPMD (2.15.0)

[error] Avoid using static access to class '\App\Services\ComposeDiff' in method 'persistCompose'. (undefined)

(StaticAccess)